pkg/network/ebpf/c/co-re/tracer-fentry.c

```diff
@@ -21,10 +21,18 @@ BPF_PERCPU_HASH_MAP(udp6_send_skb_args, u64, u64, 1024)
 BPF_PERCPU_HASH_MAP(udp_send_skb_args, u64, conn_tuple_t, 1024)
 
 #define RETURN_IF_NOT_IN_SYSPROBE_TASK(prog_name)           \
-    if (!event_in_task(prog_name)) {                        \
+    u64 task_pid_tgid = get_pid_tgid(prog_name);            \
+    if (task_pid_tgid == 0) {                               \
         return 0;                                           \
     }
 
+#define SET_PID_TGID_OR_RETURN(prog_name, pid_tgid)         \
```

Comment:
Not a fan of this macro. I think it would be better to just replace calls to `bpf_get_current_pid_tgid` with `get_pid_tgid` (or `get_sysprobe_ns_pid_tgid`, and keep using the `RETURN_IF_NOT_IN_SYSPROBE_TASK` macro separately.

Reply:
Yes, but that should be ok. I think the resulting code is clearer, and lets us use `get_pid_tgid` as a first option instead of figuring out how to incorporate this macro into our code flow.